### DISCUSSION TITLE
Please add support for loading TBL files and make it easier to search for text in hex files.

### DISCUSSION THREAD

**Discussion by Eriquin** [WerWolv/ImHex#1304] [Feedback]
The editor is very useful and complete, but I noticed that there is no support for loading a table next to the file we are editing. The function of loading a TBL file would be of great help. And please make the search engine simpler and more functional when trying to find things like words, texts, etc. in the file. Thank you very much in advance 😁

**Comment by WerWolv** [comment]
Hey
ImHex supports table files already through Custom Encodings. 
![image](https://github.com/WerWolv/ImHex/assets/10835354/b78ebf55-915c-4a70-b4f1-c5ee7cb92417)

Searching through that data however is not yet possible as of right now.

**Comment by paxcut** [comment]
Currently ImHex allows you to load up to two decoding columns. Both of them are files that users can edit or add their own. Is the functionality of a TBL different than the one from a encoding/decoding set? If so what specific functionality are you requesting? The more details you provide the easier would be to decide if the requested feature is already implemented or not.

Simplicity usually comes at the price of functionality. Balancing the twp concepts is tricky at best. Also the perception of simplicity varies among people. If you have specific features you want us to work on please list them.

**Comment by Eriquin** [comment]
Hello, thank you so much for fast answering me and sorry my bad English xD

I do some small translation or file station projects and I really need an
editor that supports uploading TBL files.

For example: when opening a file in a hex editor the texts and characters
are all scrambled or hard to find, so I create a TBL file in TableManager
(or any other software) and load that table into the previously loaded
file, The table makes the texts and characters better organized without me
having to do it manually right in the file. I don't know if I was very
clear, but basically uploading a TBL file makes editing and organizing the
texts a lot easier.

Thank you again for your attention and response :)

Em dom, 10 de set de 2023 04:35, paxcut ***@***.***> escreveu:

**Comment by paxcut** [comment]
is this what you have in mind?
![image](https://github.com/WerWolv/ImHex/assets/53811119/617d1a9c-bf76-49eb-916f-ee2afd8cb5d5)

**Comment by Eriquin** [comment]
I found a perfect example to explain, please look at this video, only from
the time 3:45 to 4:17. This stepa created a table and uploaded it to the
file. Sorry, I haven't found a better example, but this one explains well
what I mean, you'll notice what the table does with the file

https://youtu.be/ynzORDUZgo0?si=AJdQ94_DQ0xdtQcT


I'd like to ask; is it possible to add the hexadecimal file preview in only
1 Byte as a feature in your software?











Em dom, 10 de set de 2023 05:01, paxcut ***@***.***> escreveu:

**Comment by paxcut** [comment]
What you show in the video ids exactly what I show in the image. You are free to edit the encoding files and when imported like WerWolv showed, the right column will also change according to the new encoding. I think we are simply talking about the same thing using two different names. An encoding file is a lookup table that translates binary values into characters. 

Maybe the issue is with the format the encoding takes. Instead of using the location of the entry in the table as the binary value it will translate the encoding file explicitly lists both the binary and the character it will translate to. It should be easy to switch from one format to the other.

Currently ImHex supports arbitrary grouping of the binary values in the hex display . Simply move the slider shown in the picture to change the number of columns and use the drop down menu next to it to select the groupings
![image](https://github.com/WerWolv/ImHex/assets/53811119/4afbc201-4b54-4d65-9ce0-e32c86ac618e)

**Comment by Eriquin** [comment]
Oh man, so that's exactly what I was needing :) Thank you so much for your
time! I apologize for any inconvenience I may have caused. This is really
going to change me a lot

Em dom, 10 de set de 2023 05:31, paxcut ***@***.***> escreveu:
